*: bazel hackweek

irfansharif opened the pull request.

irfansharif committed:
*: bazel-ify attempt

Should let us use:

  make bazel-clean
  make bazel-generate

(I'm not checking in the generated bazel build files just yet. They'll
come in subsequent commits.)

*: check in individual BUILD.bazel files

This still does not fully work, some of the include paths (relative to
the repo root) don't pick up the right C/C++ headers correctly. In the
next commit we'll hardcode them to (my) local paths.

*: manual include path resolution

The relative path directives don't seem to work, so we pin them here.

*: get execgen working

*: get optgen+stringer working

*: pick up libproto directives

*:introduce build targets for c-deps

*: keep gen code

*: run autofmt

*: disable global proto for deps

*: point to latest rules_go

*: libroachccl + engineccl

*: proj stuff

*: point to different go-libedit

*: updating all BUILD files after rebase

*: remove unnecessary keeps

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55258)
<!-- Reviewable:end -->

irfansharif committed:
vendor: bazel-ifying go-libedit

vendor: bazel-ify vendor

vendor: pick up vendor bazil changes

irfansharif committed:
c-deps: pick up rocksdb+jemalloc changes needed for bazel

rocksdb: elide usage of __DATE__
jemalloc: fix JEMALLOC_MUTEX_INIT_CB to only be set if OSS_PINLOCK is false

Release note: None

irfansharif committed:
cdeps + cleanup

Release note (<category, see below>): <what> <show> <why>

irfansharif commented:
Subsumed by #55687.

irfansharif commented:
(Intentionally not deleting the branch here on GitHub, we have a summary doc that holds on to these references).
